gpMgmt/bin/gpexpand

```diff
@@ -2050,18 +2050,26 @@ class gpexpand:
         And prompt user to set a proper value via gpconfig before expansion.
         """
 
-        conn = dbconn.connect(_gp_expand.dburl, encoding='UTF8')
-        # get GUC value: gp_interconnect_proxy_addresses
-        sql = "show gp_interconnect_proxy_addresses;"
-        cursor = dbconn.execSQL(conn, sql)
-        icproxy_addresses_string = cursor.fetchone()[0].strip()
-        cursor.close()
-        # get GUC value: gp_interconnect_type
-        sql = "show gp_interconnect_type;"
-        cursor = dbconn.execSQL(conn, sql)
-        ic_type = cursor.fetchone()[0].strip()
-        cursor.close()
-        conn.close()
+        with closing (dbconn.connect(_gp_expand.dburl, encoding='UTF8')) as conn:
+            # get GUC value: gp_interconnect_proxy_addresses
+            try:
+                sql = "show gp_interconnect_proxy_addresses;"
+                cursor = dbconn.execSQL(conn, sql)
+                icproxy_addresses_string = cursor.fetchone()[0].strip()
+                cursor.close()
+            except DatabaseError as ex:
+                # If the error indicates that the GUC "gp_interconnect_proxy_addresses" is not recognized,
+                # it means the cluster is not using IC Proxy mode (or the feature is not enabled).
+                # In this case, we simply return without raising an error.
+                if re.search("unrecognized configuration parameter", ex.__str__()):
```

Comment:
Can we check for ERRCODE_UNDEFINED_OBJECT but not for substring?

For example, psql can show error code
postgres=# \set VERBOSITY verbose
postgres=# show gp_interconnect_proxy_addresses1;
ERROR:  42704: unrecognized configuration parameter "gp_interconnect_proxy_addresses1"
LOCATION:  GetConfigOptionByName, guc.c:8206
postgres=#

Reply:
It looks like **pygresql**, which is used by `gpexpand`, does not return the error code, unlike `psycopg2`. I couldn't find anything similar among its attributes:  

```python
['__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', 
 '__getitem__', '__getslice__', '__hash__', '__init__', '__module__', '__new__', 
 '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', 
 '__sizeof__', '__str__', '__subclasshook__', '__unicode__', '__weakref__', 
 'args', 'message']
```

So, I don't see a way to extract the error code without changing the request execution mechanism =(